metricbeat/module/vsphere/host/host.go

```diff
@@ -86,89 +116,138 @@ func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
 
 	defer func() {
 		if err := v.Destroy(ctx); err != nil {
-			m.Logger().Debug(fmt.Errorf("error trying to destroy view from vshphere: %w", err))
+			m.Logger().Errorf("error trying to destroy view from vshphere: %w", err)
 		}
 	}()
 
 	// Retrieve summary property for all hosts.
 	var hst []mo.HostSystem
-	err = v.Retrieve(ctx, []string{"HostSystem"}, []string{"summary"}, &hst)
+	err = v.Retrieve(ctx, []string{"HostSystem"}, []string{"summary", "network", "name", "vm", "datastore"}, &hst)
 	if err != nil {
 		return fmt.Errorf("error in Retrieve: %w", err)
 	}
 
-	for _, hs := range hst {
+	// Create a performance manager
+	perfManager := performance.NewManager(c)
+
+	// Retrieve metric IDs for the specified metric names
+	metrics, err := perfManager.CounterInfoByName(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve metrics: %w", err)
+	}
+
+	// Retrieve only the required metrics
+	requiredMetrics := make(map[string]*types.PerfCounterInfo)
+
+	for _, name := range metricNames {
+		metric, exists := metrics[name]
+		if !exists {
+			m.Logger().Warnf("Metric %s not found", name)
+			continue
+		}
+		requiredMetrics[name] = metric
+	}
+
+	metricIDs := make([]types.PerfMetricId, 0, len(requiredMetrics))
+	for _, metric := range requiredMetrics {
+		metricIDs = append(metricIDs, types.PerfMetricId{
+			CounterId: metric.Key,
+		})
+	}
+
+	pc := property.DefaultCollector(c)
+	for i := range hst {
+		assetNames, err := getAssetNames(ctx, pc, &hst[i])
+		if err != nil {
+			m.Logger().Errorf("Failed to retrieve object from host: %w", err)
```

Comment:
Can we deduce the host's name from the logged error value? If not, consider adding it to the log as well.